design/sequence/userclient-cli.puml

```diff
@@ -0,0 +1,35 @@
+@startuml
+
+Actor User
+
+box UserClient 
+    participant ":JobAdder" as JobAdder
+    participant ":UserCLIHandler" as UserClientCLIHandler
+    participant "userClientConfig:UserConfig" as UserClientConfig
+    participant ":ServerProxy" as ServerProxy
+    participant "userInputConfig:UserConfig" as UserInputConfig
+    participant ":ServerCommand" as ServerCommand
+end box
+
+User -> JobAdder ** : Raw user input
+activate JobAdder
+JobAdder -> UserClientCLIHandler ** : Create
+UserClientCLIHandler -> UserClientConfig ** : Create from file
+JobAdder -> ServerProxy ** : Create
+JobAdder -> UserClientCLIHandler ++ : Raw user input
+UserClientCLIHandler -> UserInputConfig ** : Create from raw user input
+UserClientCLIHandler -> UserClientConfig : Override with UserInputConfig
+UserClientCLIHandler -> ServerCommand ** : Create from UserClientConfig
+JobAdder <-- UserClientCLIHandler -- : ServerCommand
+destroy UserInputConfig
+JobAdder -> ServerProxy ++ : ServerCommand
+... ServerProxy sends ServerCommand to server, receives ServerResponse ...
+JobAdder <-- ServerProxy -- : ServerResponse
+User <-- JobAdder -- : ServerResponse
+destroy UserClientCLIHandler
+destroy UserClientConfig
+destroy ServerProxy
+destroy ServerCommand
+destroy JobAdder
+
```

Comment:
I am not entirely sure why we need to destroy all classes after running a command. Other than that, this looks good to me.

Reply:
Those are destroyed because at this point the user client will also exit I think.

Reply:
Destroy does not mean that we are going to actively destroy these objects.
Their destruction is simply a consequence of the process terminating.
When a process ends all of its data is (typically) destroyed.
For a Python process this includes all objects inside that process.

Reply:
Basically it's just to show that these objects do not stay on the system after our program is done.